sample/src/Main.fs

```diff
@@ -1,30 +1,43 @@
 ﻿module Main
 
+open Browser.Types
 open Lit
 open Fable.Haunted
 open Fable.Haunted.Types
 
+type EventTarget with
+    member this.Value = (this :?> HTMLInputElement).value
+
 type Msg =
     | Increment
     | Decrement
     | Reset
 
-
-let private another
+let private inner_component
     (props: {| sample: string
                property: string option |})
     =
-    html $"""<div>An inner component! {props.sample} - {props.property}</div>"""
+    html $"""<p>An inner component! {props.sample} - {props.property}</p>"""
 
 // defineComponent registers a Custom Element so you don't need to actually
 // call this function inside any component, you can use the component itself
-defineComponent "inner-component" (Haunted.Component(another, {| observedAttributes = [| "sample" |] |}))
+defineComponent "inner-component" (Haunted.Component(inner_component, {| observedAttributes = [| "sample" |] |}))
+
+[<VirtualComponent>]
+let virtual_component (props: {| sample: string |}) =
```

Comment:
Interesting, I had `Haunted.Virtual(renderFn)` which does the same I guess but this looks more ergonomic I like it

Reply:
We had something similar with React.functionComponent but unfortunately it has some caveats in F#. The main one is when you try to have generic components (e.g. a list of 'T items) because F# doesn't accept a generic value only functions. Some users just add a generic argument, but this becomes a function in disguise that recreatea the component every time.